content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,63 @@
+---
+title: Saving
+---
+
+GitHub's save patterns help developers to store and update their configurations for different features throughout GitHub's various interfaces. These changes should be represented in the UI accurately, quickly, and obviously. Their behavior should be easy to reason about and inspire confidence and trust.
+
+All configuration flows should be punctuated with CTAs that feature a logical term such as "Create", "Save", "Delete", or "Update" (e.g. "Create" a new repository, "Add" an SSH key to your profile, "Save" security preferences, "Update" a repository's description, "Delete" an old email address). 
+
+Individual settings that are not changed via discrete flows should minimize the required UI by saving configuration changes as they are made (e.g. changing your primary email address, adding an assignee to an issue, unsubscribing from a notification type, following a user, watching a repository).
+
+## There are two primary patterns that should be used for saving user input on GitHub:
+### Save and route
+This pattern is primarily used for creation and deletion of new entities such as repositories, organizations, or user accounts. At the end of the flow, the user should be routed to the entity they just created. In some cases such as forking or using a template, we may need to hold the user at a waiting page while the entity is generated. In these cases, we should make the user feel confident that something is happening and that their configuration details will not be lost. In some cases, such as deleting a repository, it may make sense to be explicit that the interaction has occurred by using a [Flash alert](https://primer.style/css/components/alerts).
+
```

Comment:
Will this have a visual example when the visuals are agreed? (same question for optimistic update section)

Reply:
Since the Flash alert page has the actual details, I figured I'd leave it to that? I'm certainly open to it, but I didn't think there was a clear fit. It just felt like any visual detail would be better provided by the components' pages themselves.

If you've got ideas, I'm certainly open to it!

Reply:
Do we want to be specific and opinionated about the anatomy of a post-save flash alert? 

Do we always use an icon vs not? What kind of style do we use for success? (depending on the page, you can find default flashes for indicating success)

If we have a point of view, we should document it and provide examples.